zksync/test/weth.test.ts

```diff
@@ -0,0 +1,109 @@
+import { expect } from "chai";
+import { Wallet, Provider } from "zksync-web3";
+import * as hre from "hardhat";
+import { ethers } from "ethers";
+import { L2WethFactory } from "../typechain/L2WethFactory";
+import { L2Weth} from "../typechain/L2Weth";
+import { L2WethBridgeFactory } from "../typechain/L2WethBridgeFactory";
+import { L2WethBridge } from "../typechain/L2WethBridge";
+import { Deployer } from "@matterlabs/hardhat-zksync-deploy";
+
+const richAccount =   {
+    address: "0x36615Cf349d7F6344891B1e7CA7C72883F5dc049",
+    privateKey: "0x7726827caac94a7f9e1b160f7ea819f172f7b6f9d2a97f992c38edeab82d4110",
+};
+
+const eth18 = ethers.utils.parseEther("18");
+
+describe("WETH token & WETH bridge", function () {
+  let provider = new Provider(hre.config.networks.localhost.url);
+  let wallet = new Wallet(richAccount.privateKey, provider);
+  let wethToken: L2Weth;
+  let wethBridge: L2WethBridge;
+
+  before("Deploy token and bridge", async function () {
+      const deployer = new Deployer(hre, wallet);
+      const wethTokenImpl = await deployer.deploy(await deployer.loadArtifact("L2Weth"));
+      const wethBridgeImpl = await deployer.deploy(await deployer.loadArtifact("L2WethBridge"));
+      const randomAddress = ethers.utils.hexlify(ethers.utils.randomBytes(20));
+
+      const wethTokenProxy = await deployer.deploy(
+          await deployer.loadArtifact("TransparentUpgradeableProxy"), 
+          [wethTokenImpl.address, randomAddress, "0x"]
+      );
+      const wethBridgeProxy = await deployer.deploy(
+          await deployer.loadArtifact("TransparentUpgradeableProxy"), 
+          [wethBridgeImpl.address, randomAddress, "0x"]
+      ) as any;
+
+      wethToken = L2WethFactory.connect(wethTokenProxy.address, wallet);
+      wethBridge = L2WethBridgeFactory.connect(wethBridgeProxy.address, wallet);
+
+      await wethToken.initialize("Wrapped Ether", "WETH");
+      await wethToken.initializeV2(wethBridge.address, randomAddress);
+
+      await wethBridge.initialize(randomAddress, randomAddress, wethToken.address);
+  });
+
+  it("Should deposit WETH by calling deposit()", async function () {
+      await wethToken.deposit({ value: eth18 }).then(tx => tx.wait());
+      expect(await wethToken.balanceOf(wallet.address)).to.equal(eth18);
+  });
+
+  it("Should deposit WETH by sending", async function () {
```

Comment:
Please add a negative test case, where sent with random calldata. It should fail, since WETH token implements `receive` but not `fallback` function.